zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -1023,39 +1091,90 @@ pub(crate) fn transaction_data_requests<P: consensus::Parameters>(
     // `lightwalletd` will return an error for `GetTaddressTxids` requests having an end height
     // greater than the current chain tip height, so we take the chain tip height into account
     // here in order to make this pothole easier for clients of the API to avoid.
-    let chain_tip_height = super::chain_tip_height(conn)?;
+    let chain_tip_height =
+        super::chain_tip_height(conn)?.ok_or(SqliteClientError::ChainHeightUnknown)?;
 
     // We cannot construct address-based transaction data requests for the case where we cannot
     // determine the height at which to begin, so we require that either the target height or mined
     // height be set.
-    let mut address_request_stmt = conn.prepare_cached(
-        "SELECT ssq.address, IFNULL(t.target_height, t.mined_height)
+    let mut spend_requests_stmt = conn.prepare_cached(
+        "SELECT
+            ssq.address,
+            IFNULL(t.target_height, t.mined_height)
          FROM transparent_spend_search_queue ssq
          JOIN transactions t ON t.id_tx = ssq.transaction_id
          WHERE t.target_height IS NOT NULL
          OR t.mined_height IS NOT NULL",
     )?;
 
-    let result = address_request_stmt
-        .query_and_then([], |row| {
+    let spend_search_rows = spend_requests_stmt.query_and_then([], |row| {
+        let address = TransparentAddress::decode(params, &row.get::<_, String>(0)?)?;
+        let block_range_start = BlockHeight::from(row.get::<_, u32>(1)?);
+        let max_end_height = block_range_start + DEFAULT_TX_EXPIRY_DELTA + 1;
+        Ok::<TransactionDataRequest, SqliteClientError>(
+            TransactionDataRequest::TransactionsInvolvingAddress {
+                address,
+                block_range_start,
+                block_range_end: Some(std::cmp::min(chain_tip_height + 1, max_end_height)),
+                request_at: None,
+                tx_status_filter: TransactionStatusFilter::Mined,
+                output_status_filter: OutputStatusFilter::All,
+            },
+        )
+    })?;
+
+    // Perform ephemeral address checks only for addresses that do not have an unexpired
+    // transaction associated with them in the database.
```

Comment:
This wording could be clarified; the expiry height is serving a seondary purpose of re-allowing checks on ephemeral addresses for which their unexpired transactions were mined prior to expiry, once those transactions are old enough to have expired if they were not mined. They are not "expired transactions", and thus can be interpreted as "an unexpired transaction associated with [the ephemeral address]", but that's not the kind of expiry we care about here.